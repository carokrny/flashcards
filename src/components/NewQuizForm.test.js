import React from "react";
import { Provider } from "react-redux";
import renderer from 'react-test-renderer';
import configureStore from "redux-mock-store";
import NewQuizForm from "./NewQuizForm";

const mockStore = configureStore([]);

describe('<NewQuizForm /> component', () => {
    let store; 
    let component;
    
    beforeEach(() => {
        store = mockStore({
            topics: {
                topics: {
                    '123': {
                        id: '123',
                        name: 'example topic1',
                        icon: 'icon url1',
                        quizIds: ['456'],
                    }
                } 
            },
            quizzes: {
                quizzes: {
                    '456': {
                        id: '456',
                        topicId: '123',
                        name: 'quiz 1 for example topic1',
                        cardIds: ['1', '2', '3'],
                    }
                }
            }
        })

        component = renderer.create(
            <Provider store={store}>
                <NewQuizForm />
            </Provider>
        )
    }),

    describe('rendering', () => {
        it('should render the component without crashing', () => {
            expect(component.toJSON()).toBeTruthy();
        }),
        
        it('should render with given state from Redux store', () => {
            expect(component.toJSON()).toMatchSnapshot();
        }),

        it('should render a form', () => {
            expect(component.root.findByType('form')).toBeTruthy();
        })
    })

});
