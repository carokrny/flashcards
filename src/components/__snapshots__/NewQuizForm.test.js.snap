// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewQuizForm /> component rendering should render with given state from Redux store 1`] = `
<section>
  <h1>
    Create a new quiz
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input
      id="quiz-name"
      onChange={[Function]}
      placeholder="Quiz Title"
      value=""
    />
    <select
      id="quiz-topic"
      onChange={[Function]}
      placeholder="Topic"
    >
      <option
        value=""
      >
        Topic
      </option>
      <option
        value="123"
      >
        example topic1
      </option>
    </select>
    <div
      className="actions-container"
    >
      <button
        onClick={[Function]}
      >
        Add a Card
      </button>
      <button>
        Create Quiz
      </button>
    </div>
  </form>
</section>
`;
